core/services/job/orm.go

```diff
@@ -45,70 +45,93 @@ var (
 	DismissError(ctx context.Context, errorID int64) error
-	FindSpecError(id int64, qopts ...pg.QOpt) (SpecError, error)
+	FindSpecError(ctx context.Context, id int64) (SpecError, error)
 	Close() error
-	PipelineRuns(jobID *int32, offset, size int) ([]pipeline.Run, int, error)
+	PipelineRuns(ctx context.Context, jobID *int32, offset, size int) ([]pipeline.Run, int, error)
 
-	FindPipelineRunIDsByJobID(jobID int32, offset, limit int) (ids []int64, err error)
-	FindPipelineRunsByIDs(ids []int64) (runs []pipeline.Run, err error)
-	CountPipelineRunsByJobID(jobID int32) (count int32, err error)
+	FindPipelineRunIDsByJobID(ctx context.Context, jobID int32, offset, limit int) (ids []int64, err error)
+	FindPipelineRunsByIDs(ctx context.Context, ids []int64) (runs []pipeline.Run, err error)
+	CountPipelineRunsByJobID(ctx context.Context, jobID int32) (count int32, err error)
 
-	FindJobsByPipelineSpecIDs(ids []int32) ([]Job, error)
-	FindPipelineRunByID(id int64) (pipeline.Run, error)
+	FindJobsByPipelineSpecIDs(ctx context.Context, ids []int32) ([]Job, error)
+	FindPipelineRunByID(ctx context.Context, id int64) (pipeline.Run, error)
 
-	FindSpecErrorsByJobIDs(ids []int32, qopts ...pg.QOpt) ([]SpecError, error)
-	FindJobWithoutSpecErrors(id int32) (jb Job, err error)
+	FindSpecErrorsByJobIDs(ctx context.Context, ids []int32) ([]SpecError, error)
+	FindJobWithoutSpecErrors(ctx context.Context, id int32) (jb Job, err error)
 
-	FindTaskResultByRunIDAndTaskName(runID int64, taskName string, qopts ...pg.QOpt) ([]byte, error)
-	AssertBridgesExist(p pipeline.Pipeline) error
+	FindTaskResultByRunIDAndTaskName(ctx context.Context, runID int64, taskName string) ([]byte, error)
+	AssertBridgesExist(ctx context.Context, p pipeline.Pipeline) error
+
+	DataSource() sqlutil.DataSource
+	WithDataSource(source sqlutil.DataSource) ORM
 }
 
 type ORMConfig interface {
 	DatabaseDefaultQueryTimeout() time.Duration
 }
 
 type orm struct {
-	q           pg.Q
+	db          sqlutil.DataSource
 	keyStore    keystore.Master
 	pipelineORM pipeline.ORM
 	lggr        logger.SugaredLogger
-	cfg         pg.QConfig
 	bridgeORM   bridges.ORM
 }
 
 var _ ORM = (*orm)(nil)
 
-func NewORM(db *sqlx.DB, pipelineORM pipeline.ORM, bridgeORM bridges.ORM, keyStore keystore.Master, lggr logger.Logger, cfg pg.QConfig) *orm {
+func NewORM(db sqlutil.DataSource, pipelineORM pipeline.ORM, bridgeORM bridges.ORM, keyStore keystore.Master, lggr logger.Logger) *orm {
 	namedLogger := logger.Sugared(lggr.Named("JobORM"))
 	return &orm{
-		q:           pg.NewQ(db, namedLogger, cfg),
+		db:          db,
 		keyStore:    keyStore,
 		pipelineORM: pipelineORM,
 		bridgeORM:   bridgeORM,
 		lggr:        namedLogger,
-		cfg:         cfg,
 	}
 }
+
 func (o *orm) Close() error {
 	return nil
 }
 
-func (o *orm) AssertBridgesExist(p pipeline.Pipeline) error {
+func (o *orm) DataSource() sqlutil.DataSource {
+	return o.db
+}
+
+func (o *orm) WithDataSource(tx sqlutil.DataSource) ORM { return o.withDataSource(tx) }
```

Comment:
is this supposed to be a transaction or any data source? if the former, rename the method, if the latter, rename the var to ds